src/main/scala/zio/prelude/DistributiveMultiply.scala

```diff
@@ -0,0 +1,53 @@
+package zio.prelude
+
+import zio.prelude.coherent.DistributiveMultiplyEqual
+import zio.test.TestResult
+import zio.test.laws.{ Lawful, Laws }
+
+trait DistributiveMultiply[A, +Addition[x] <: Associative[x], +Multiplication[x] <: Associative[x]]
+    extends AddMultiplyShape[A, Addition, Multiplication]
+
+object DistributiveMultiply extends Lawful[DistributiveMultiplyEqual] {
+
+  /**
+   * The left distributivity law states that for operators `+` and `*`, for all
+   * values `a1`, `a2`, and `a3`, the following must hold:
+   *
+   * {{{
+   * a1 * (a2 + a3) === (a1 * a2) + (a1 * a3)
+   * }}}
+   */
+  val leftDistributivityLaw: Laws[DistributiveMultiplyEqual] =
```

Comment:
Why are these laws introduced here instead of in AddMultiply ("Distributive")? What useful (non-esoteric) structures are there without the distributive law?

Reply:
> What useful (non-esoteric) structures are there without the distributive law?
If there are any, I have no idea of their existence. (I'm no mathematician, though :smile: )
But that was not the reason why I introduced both `AnnihilatingZero` and `DistributiveMultiply`.

The traditional algebraic structure _Semiring_ requires (among others) these two, independent, laws: (1) annihilating zero and (2) distributing multiply. That's why I introduced two independent type classes: `AnnihilatingZero` and `DistributiveMultiply`.

Do we want to express it in another way? What would be an alternative? `AnnihilatingZero` inheriting from `DistributiveMultiply`? Or vice versa? :thinking:

Btw, I intentionally named it verbosely `DistributiveMultiply`, so that it's distinct from `DistributiveJoinMeet` (for Lattices).